v2/AdminCore.Services/DashboardService.cs

```diff
@@ -163,14 +165,64 @@ private static EmployeeSnapshotDto CreateEmployeeSnapshotFromContract(Contract c
     {
       return new EmployeeSnapshotDto()
       {
-        ClientName = contract.Team.Client.ClientName,
-        Description = contract.Employee.Events.First().EventType.Description,
-        TeamName = contract.Team.TeamName,
-        TeamId = contract.TeamId,
         Email = contract.Employee.Email,
         EmployeeId = contract.EmployeeId,
-        Name = $"{contract.Employee.Forename} {contract.Employee.Surname}"
+        Forename = contract.Employee.Forename,
+        Surname = contract.Employee.Surname
       };
     }
+
+    private static Expression<Func<Client, bool>> QueryClientsWithContractsForEmployeeId(IList<int> teamIds, DateTime date)
+    {
+      return client =>
+        client.Teams.Any(team => teamIds.Contains(team.TeamId) &&
+          team.Contracts.Any(contract =>
+            DateService.ContractIsActiveDuringDate(contract, date)));
+    }
+
+    private ClientSnapshotDto BuildClientSnapshot(Client client, DateTime date)
+    {
+      var clientSnapShot = _mapper.Map<ClientSnapshotDto>(client);
+      clientSnapShot.Teams = client.Teams.Select(clientTeam => BuildTeamSnapshot(clientTeam, date)).ToList();
+      return clientSnapShot;
+    }
+
+    private TeamSnapshotDto BuildTeamSnapshot(Team team, DateTime date)
+    {
+      var teamSnapshot = _mapper.Map<TeamSnapshotDto>(team);
+      teamSnapshot.Employees = team.Contracts.Select(contract => BuildEmployeeSnapshot(contract.Employee, date)).ToList();
+      return teamSnapshot;
+    }
+
+    private EmployeeSnapshotDto BuildEmployeeSnapshot(Employee employee, DateTime date)
+    {
+      var employeeSnapshot = _mapper.Map<EmployeeSnapshotDto>(employee);
+      employeeSnapshot.Location = GetLocationFromEmployee(employee, date);
+      return employeeSnapshot;
+    }
+
+    private string GetLocationFromEmployee(Employee employee, DateTime date)
+    {
+      var employeeEvents = employee.Events.Where(evnt =>
+        DateService.EventContainsEventDatesThatHappenDuringAGivenDay(evnt.EventDates, date)).ToList();
+      return employeeEvents.Any() ? GetLocationFromEvent(employeeEvents.First()) : EmployeeLocationConstants.InOffice;
+    }
+
+    private static string GetLocationFromEvent(Event @event)
+    {
+      switch (@event.EventTypeId)
+      {
+        case 1:
+          return EmployeeLocationConstants.OnHoliday;
```

Comment:
Small thing, change this to AnnualLeave for consistency with the rest of the project

Reply:
OnHoliday encompasses multiple holiday types (Annual Leave, Public Holiday, Mandatory leave.. etc..) and was explicitly requested by FE